cbdr/cortex
# AR-67+68: Publish Date Sorting Kerfuffles

## Purpose:
Fixes our sorting logic for the posts feed so that if the published filter is applied the total set of posts are sorted by the `published_at` date, rather than the `created_at` date. Leaves the expected return the same on legacy cortex.

**AR-67:**
> Articles on the filtered landing pages (Discovery, Get the Job, etc.) are not in order of the date published and we would like them to be, similar to the A&R homepage.

**AR-68**
> Ensure all searches for A&R deliver results by the visible, published date on a post - NOT the create date or modified date.

## JIRA:
* https://cb-content-enablement.atlassian.net/browse/AR-67
* https://cb-content-enablement.atlassian.net/browse/AR-68

## Steps to Take On Prod
* Go to `/api/v1/posts/feed?per_page=100` (several per page to prove behavior)
* Confirm that, with the notable exception of sticky posts, the posts are in order of publish date, with the most recent being at the top

## Changes:
* Changes to setup
  * Rework ES logic for Posts

* Architectural changes
  * ^

* Migrations
  * N/A

* Library changes
  * N/A

* Side effects
  * N/A

## Screenshots
* Before
N/A

* After
N/A

## QA Links:
http://web.cortex-6.development.c66.me/ (Having deploy issues ATM)

## How to Verify These Changes
* Specific pages to visit
  * `/api/v1/posts/feed?per_page=100`

* Steps to take
  * Confirm that, with the exception of sticky posts, the posts are sorted by Published Date, with the most recent at the top.

* Responsive considerations
  * N/A

## Relevant PRs/Dependencies:
#495: Upgrading Cortex to ES 5.4.0, which this is fully compatible with

## Additional Information
N/A
